Monsters hunting persistence/src/main/java/cz/fi/muni/pa165/entity/Area.java

```diff
@@ -1,4 +1,104 @@
 package cz.fi.muni.pa165.entity;
 
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.Enumerated;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.ManyToMany;
+import javax.validation.constraints.NotNull;
+import java.util.HashSet;
+import java.util.Set;
+
+import cz.fi.muni.pa165.enums.AreaType;
+
+/**
+ *   @author Jan Gol <gol.honza@gmail.com>
+ */
+@Entity
+
 public class Area {
+    
+    	@Id
+	@GeneratedValue(strategy = GenerationType.SEQUENCE)
+	private Long id;
+
+	@NotNull
+	@Column(nullable = false, unique = true)
+	private String name;
+        
+        @Enumerated
+        private AreaType type;
+        
+        @ManyToMany()
+        private Set<Monster> monsters = new HashSet<>();
+        
+        public Area(){
+        }
+        
+        public Area(String name){
+            checkName(name);
+            this.name = name;
+        }
+        
+        public Long getId() {
+		return id;
+	}
+
+	public void setId(Long id) {
+		this.id = id;
+        }
+
+	public String getName() {
+		return name;
+	}
+
+	public void setName(String name) {
+		checkName(name);
+		this.name = name;
+	}
+        
+        public AreaType getType() {
+                return type;
+        }
+
+        public void setType(AreaType type) {
+                this.type = type;
+        }
+        
+        public Set<Monster> getAreas() {
+                return monsters;
```

Comment:
I guess that name of this method was ment to be ```getMonsters()``` please change it. Also, return the collection as unmofiable